src/lib.rs

```diff
@@ -1,14 +1,259 @@
-pub fn add(left: usize, right: usize) -> usize {
-    left + right
+#![no_std]
+#![warn(unreachable_pub)]
```

Comment:
Did you consider matching the linting config from the main rustls repo?

Over there we treat `deny` `unreachable_pub` and also include some others like `missing_docs`, `clippy::use_self` etc. I think some of those might be worthwhile here too.

Reply:
I kind of wanted to start from scratch because there are some downsides to cargo-culting around the setup from older repos. A bunch of the lints that we are setup to `allow` in rustls seem like they are no longer enabled by default, for example, and all other things being equal I'd rather have smaller lint deviants from default.

In general for most of these things I prefer avoiding `deny` and using `warn` instead while denying warnings in CI only.

So this is more like my "native" minimal lint setup. Extending this with more stuff could make sense but is low on my priority list relative to finding a good API that can serve all the crates.

Reply:
That makes sense to me, thanks for explaining :+1: